projects/ngx-sub-form/src/lib/ngx-sub-form.component.ts

```diff
@@ -21,17 +30,19 @@ export abstract class NgxSubFormComponent<ControlInterface, FormInterface = Cont
     return this.mapControls(ctrl => ctrl.value) as Required<FormInterface>;
   }
 
-  public get formGroupErrors(): null | Partial<ControlMap<FormInterface, ValidationErrors | null>> {
+  public get formGroupErrors(): null | Partial<
+    ControlMap<FormInterface, ValidationErrors | null> & { formGroup: ValidationErrors }
+  > {
     const errors = this.mapControls<ValidationErrors | null, ControlMap<FormInterface, ValidationErrors | null>>(
       ctrl => ctrl.errors,
       ctrl => ctrl.invalid,
     );
 
-    if (!errors || !Object.keys(errors).length) {
+    if (!this.formGroup.errors && (!errors || !Object.keys(errors).length)) {
       return null;
     }
 
-    return errors;
+    return Object.assign({}, this.formGroup.errors ? { formGroup: this.formGroup.errors } : {}, errors);
```

Comment:
Should we have both (always)?

Ex

`{ formGroup: this.formGroup.errors, fields: errors }`

Reply:
Yea I wondered this, but kinda decided against it - I feel like it is rather rare to need to do a form level validation, so I'd prefer to keep the templates simpler for the controls. An example of how this would look is


```html
<input type="text" placeholder="Password" [formControlName]="formControlNames.password" />
<mat-error *ngIf="formControlErrors?.password?.length">Password too short</mat-error>

<input type="text" placeholder="Repeat Password" [formControlName]="formControlNames.passwordRepeat" />
<mat-error *ngIf="formControlErrors?.formGroup?.passwordsDontMatch">Passwords do not match</mat-error>
```